cmd/gitserver/server/server_grpc.go

```diff
@@ -168,3 +170,17 @@ func (gs *GRPCServer) Search(req *proto.SearchRequest, ss proto.GitserverService
 		},
 	})
 }
+
+func (gs *GRPCServer) ReposStats(ctx context.Context, req *proto.ReposStatsRequest) (*proto.ReposStatsResponse, error) {
+	b, err := gs.Server.readReposStatsFile(filepath.Join(gs.Server.ReposDir, reposStatsName))
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to read %s: %v", reposStatsName, err.Error())
+	}
+
+	var stats *protocol.ReposStats
+	if err := json.Unmarshal(b, &stats); err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to unmarshal %s: %v", reposStatsName, err.Error())
+	}
```

Comment:
nit: `err.Error()` is a string value, so we can use the more specific `%s`.

```suggestion
	if err != nil {
		return nil, status.Errorf(codes.Internal, "failed to read %s: %s", reposStatsName, err.Error())
	}

	var stats *protocol.ReposStats
	if err := json.Unmarshal(b, &stats); err != nil {
		return nil, status.Errorf(codes.Internal, "failed to unmarshal %s: %s", reposStatsName, err.Error())
	}
```